specs/core/0_beacon-chain.md

```diff
@@ -1236,6 +1253,32 @@ def process_deposit(state: BeaconState, deposit: Deposit) -> None:
     """
     deposit_input = deposit.deposit_data.deposit_input
 
+    # Should equal 8 bytes for deposit_data.amount +
+    #              8 bytes for deposit_data.timestamp +
+    #              176 bytes for deposit_data.deposit_input
+    # It should match the deposit_data in the eth1.0 deposit contract
+    serialized_deposit_data = ssz_serialize(deposit.deposit_data)
+    # Deposits must be processed in order
+    assert deposit.index == state.deposit_index
+    
+    # Verify the Merkle branch
+    merkle_branch_is_valid = verify_merkle_branch(
+        leaf=hash(serialized_deposit_data),
+        branch=deposit.branch,
+        depth=DEPOSIT_CONTRACT_TREE_DEPTH,
+        index=deposit.index,
+        root=state.latest_eth1_data.deposit_root
+    )
+    if not merkle_branch_is_valid:
```

Comment:
I think we should actually throw here with an `assert` rather than fail silently.
Block should be invalid if the block producer provided an invalid data/branch for the deposit_index/root